addons/stock/models/stock_move.py

```diff
@@ -686,11 +686,16 @@ def _assign_picking(self):
         type (moves should already have them identical). Otherwise, create a new
         picking to assign them to. """
         Picking = self.env['stock.picking']
-        for move in self:
-            recompute = False
+        groupby = ['group_id', 'location_id', 'location_dest_id', 'picking_type_id']
+        grouped_moves = self.env['stock.move'].read_group([('id', 'in', self.ids)], groupby, groupby)
+        for group in grouped_moves:
+            moves = self.env['stock.move'].search(group['__domain'])
```

Comment:
self is already a recordset
now read_group will make database requests
maybe you could collections.groupby here

Reply:
anyway i don't get the need of grouping here

Reply:
ok it's to avoid a search by move
-> make this in another commit